### PR TITLE
Add h1 and pdf title

### PR DESCRIPTION
**Pull request by gmrabian**
# Description

Our PDFs were failing two accessibility findings: h1 and document title. This resolves both.

- Broke out the h1 title into its own component so it can be used independently of the PageInfo component.
- Modified Redux state to pull the form's state name from the path (applies to viewing/editing form)
- Added a title to the printable report as well as the h1 object. Pulls state name and form year from query params (Since this opens in a separate tab couldn't reuse state)
- Created unique styling for title appearing independently vs with the PageInfo (different left margins)
- Added a query param check to the redux form year update (fixes bug where a form from 2021 displays with year 2020 on the print page)

If you have suggestions for improving this, please let me know!!

[Ticket](https://qmacbis.atlassian.net/browse/MDCT-153?atlOrigin=eyJpIjoiMmIyZGEyNjMwY2FjNDA1MWE1N2I1ZDM5ZDk4MWQ1ZjMiLCJwIjoiaiJ9)

Any tests I can add to this/will this stay the same in v3?

## How to test

1. Change into the react directory under `frontend/react`
2. `nvm use 14.4`
3. Run the following command to create a local environment that points to the deployed dev api `API_POSTGRES_URL=https://api.mdctcartsdev.cms.gov/ ./env.sh && cp env-config.js public/`
4. Start react locally with `npm run start`
Then you can login and do everything from localhost:3000 (incognito recommended)

Things to look for: State name shows up properly when switching between reports in admin view.
When I select print entire form, the correct title displays on the preview. When I click print on the preview, the document has a title in the format `{State Name} CARTS FY{XXXX} Report`

## Code authors checklist
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests. [Testing Doc](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research)
- [x] Necessary analytics were added, or no new analytics were required
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned the PR to myself

## Reviewers Checklist (Two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review 

## Assignee 
- [ ] I have closed the PR after the review and necessary changes (squashing preferred)

### CODE REVIEW

**Review by karla-vm — APPROVED** [review]

**Comment by gmrabian** [comment]
Oh that's good to know @karla-vm  I didn't think about that. Let me put in a quick fix.

File: frontend/react/src/components/layout/Title.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+const Title = ({ stateName, formYear, urlStateName }) => {
+  let displayStateName = stateName;
+  if (stateName === undefined && urlStateName !== undefined) {
+    displayStateName = urlStateName;
+  } else if (stateName === undefined && urlStateName === undefined) {
+    displayStateName = "";
```
Comment (BearHanded) [inline frontend/react/src/components/layout/Title.js:10]: When you're logged in as a state user, you can actually find this display name on in stateUser.name, or stateUser.currentUser.state.name.

I struggled through some of this while pairing with Karla the other day, let me know if you want another set of eyes.

File: frontend/react/src/components/sections/Print.js [hunk 2]
```diff
@@ -18,6 +21,12 @@ import axios from "../../authenticatedAxios";
  */
 const Print = ({ currentUser, state }) => {
   const dispatch = useDispatch();
+  const search = useLocation().search;
+  const stateInitials = new URLSearchParams(search).get("state");
```
Comment (BearHanded) [inline frontend/react/src/components/sections/Print.js:25]: state does not appear in the url for state users, you'll need to get that info elsewhere or anticipate an undefined

File: frontend/react/src/scss/_main.scss [hunk 3]
```diff
@@ -49,8 +49,23 @@
...
+
+    h1 {
+      font-weight: bold;
+      margin-top: 0;
+    }
+  }
+}
+
+.h1-title-report {
+  margin-top: 16px;
```
Comment (BearHanded) [inline frontend/react/src/scss/_main.scss:64]: ideally these should be referencing the padding variable. $padding * 2 will get you 16px

**Review by karla-vm — APPROVED** [review]
I'm able to see the correct state in the form and print version (as a state user)

**Review by braxex — APPROVED** [review]
Just some suggestions. Nothing blocking.

**Review by braxex — APPROVED** [review]
Looks great. Good job! 👍

**Review by braxex — APPROVED** [review]

**Review by karla-vm — APPROVED** [review]
Worked on my State User report! Looks great.

**Pull request merged**